app/scripts/services/authorization.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("AuthorizationService", function($q, Logger, $interval, DataService, APIService){
+    function AuthorizationService() {
+      this.rules = null;
+      this.intervalTimer = null;
+      this.forceReload = false;
+      this.canICreateSomeResource = false;
+      this.projectNameRules = null;
+    }
+
+    var normalizeRules = function(rules) {
+      var normalizedRules = {};
+      _.each(rules, function(rule) {
+        _.each(rule.resources, function(resource) {
+          normalizedRules[resource] = rule.verbs;
+        });
+      });
+      return normalizedRules;
+    };
+
+    AuthorizationService.prototype.getProjectRules = function(projectName) {
+      var self = this;
+      if (!self.rules || projectName !== self.projectNameRules) {
+        // Making an API call to get the rules if:
+        // - no rules are cached
+        // - when need to force update the rules
+        // - when switching between projects
+        Logger.log("AuthorizationService, loading user rules");
+        console.log("AuthorizationService, loading user rules");
+        if (self.forceReload) {
+          self.setForceReload(false);
+        }
+        if (!self.intervalTimer) {
+          // Force rules reload every 3 minutes
```

Comment:
@liggitt agreed that the interval is too short :+1: 
My point was not to hide/show the action btns/links on fly, but rather on next page to avoid any confusion. So it all come to the question if we want to do the changes (if there will be any) on the same page or on next.
Thoughts ?